user-console/src/main/java/org/pentaho/mantle/client/admin/EmailAdminPanel.java

```diff
@@ -74,26 +90,53 @@ private Widget createEmailPanel() {
     vSpacer.setHeight( "20px" );
     mailPanel.add( vSpacer );
 
-    mailPanel.add( new Label( Messages.getString( "smtpHost" ) + ":" ) );
-    smtpHostTextBox = new TextBox();
-    smtpHostTextBox.setWidth( "220px" );
-    mailPanel.add( smtpHostTextBox );
+    HorizontalPanel protocolHbox = new HorizontalPanel();
+    protocolHbox.add( new Label( Messages.getString( "connectionProtocol" ) + ":" ) );
+    protocolHbox.getElement().setId( "protocolPanel" );
+
+    SimplePanel hSpacer = new SimplePanel();
+    hSpacer.setWidth( "15px" );
+    protocolHbox.add( hSpacer );
+
+    protocolsListBox = new ListBox();
+    protocolsListBox.addItem( Messages.getString( "smtp" ), "SMTP" );
```

Comment:
Create enum class for all the protocols. These Strings are repeated in the code "SMTP", "SMTPS" and the one "graph_api". And replace everywhere.

I'd keep with the convention of using all caps  https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

Reply:
see  comment by @dcleao https://github.com/pentaho/pentaho-platform/pull/5185/files#r1081881270
I don't care enum or constant.